### PR TITLE
Fix #1658: Fix constant-folding of classOf[T] === classOf[T].

### LINKED ISSUE #1658: Incorrect constant-folding of `classOf[T] === classOf[T]`.
Constant-folding a `===` of two literal `ClassOf`s always return `false`, e.g.:
```scala
println(classOf[String] == classOf[String])
```
prints `false`.

### PR DESCRIPTION
**Pull request by sjrd**

### CODE REVIEW

**Comment by sjrd** [comment]
Review by @nicolasstucki

File: test-suite/src/test/scala/org/scalajs/testsuite/compiler/OptimizerTest.scala [hunk 1]
```diff
@@ -65,6 +65,20 @@ object OptimizerTest extends JasmineTest {
...
     }
 
+    it("must not break classOf[T] == classOf[U] - #1658") {
+      expect(classOf[String] == classOf[String]).toBeTruthy
+      expect(classOf[Int] == classOf[Int]).toBeTruthy
+      expect(classOf[Array[Int]] == classOf[Array[Int]]).toBeTruthy
+      expect(classOf[Array[String]] == classOf[Array[String]]).toBeTruthy
+
+      expect(classOf[String] == classOf[Int]).toBeFalsy
+      expect(classOf[Seq[_]] == classOf[List[_]]).toBeFalsy
```
Comment (nicolasstucki) [inline test-suite/src/test/scala/org/scalajs/testsuite/compiler/OptimizerTest.scala]: What about `classOf[Long] == classOf[Int]` and other classes that end up using the js primitive representation.

**Comment by nicolasstucki** [comment]
Thats all.

File: test-suite/src/test/scala/org/scalajs/testsuite/compiler/OptimizerTest.scala (hunk 1 above)
Comment (sjrd) [inline test-suite/src/test/scala/org/scalajs/testsuite/compiler/OptimizerTest.scala]: At the `j.l.Class` level, it's pretty irrelevant how they end up being represented in JS.

For tests relevant to JS primitive representation, see https://github.com/scala-js/scala-js/blob/master/test-suite/src/test/scala/org/scalajs/testsuite/compiler/InstanceTestsHijackedBoxedClassesTest.scala

But I can add `classOf[Int] == classOf[Long]` if you want ;)

Comment (nicolasstucki) [inline test-suite/src/test/scala/org/scalajs/testsuite/compiler/OptimizerTest.scala]: Ok.

**Comment by nicolasstucki** [comment]
LGTM

**Pull request merged**